packages/objection/src/interpreter.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  createInterpreter,
+  Condition,
+  InterpretationContext
+} from '@ucast/core';
+import { Model as ObjectionModel, QueryBuilder, QueryBuilderType, Relations } from 'objection';
+
+type QueryBuilderMethod = keyof QueryBuilder<ObjectionModel> & 'where' | 'orWhere' | 'whereNot';
+
+export class Query {
+  public query: QueryBuilder<ObjectionModel>;
+  private _method: QueryBuilderMethod;
+  private _relations: Relations;
+
+  constructor(query: QueryBuilder<ObjectionModel>, method: QueryBuilderMethod = 'where') {
+    this.query = query;
+    this._method = method;
+    this._relations = query.modelClass().getRelations();
+  }
+
+  where(field: string, operator: string, value: any) {
+    const possibleMethod = this._method + operator as QueryBuilderMethod;
+
+    if (field.includes('.')) {
```

Comment:
I'd like to have here micro optimization :) So, instead of doing includes and split later we can use `indexOf` and `slice` on string (which is more efficient):

```js
const relationNameIndex = field.indexOf('.');

if (relationNameIndex !== -1) {
  const relationName = field.slice(0, relationNameIndex);
  // the rest of your code
}
```